packages/react-api/src/hooks/useGeoJsonFeatures.js

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useCallback, useState, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { setViewportFeaturesReady } from '@carto/react-redux';
+import { debounce } from '@carto/react-core';
+import { Methods, executeTask } from '@carto/react-workers';
+import { throwError } from './utils';
+
+export default function useGeoJsonFeatures({
+  source,
+  viewport,
+  uniqueIdProperty = 'cartodb_id',
+  debounceTimeout = 250
+}) {
+  const dispatch = useDispatch();
+  const [isGeoJsonLoaded, setGeoJsonLoaded] = useState(false);
+  const debounceIdRef = useRef(null);
+
+  const clearDebounce = () => {
+    if (debounceIdRef.current) {
+      clearTimeout(debounceIdRef.current);
+    }
+    debounceIdRef.current = null;
+  };
+
+  const sourceId = source?.id;
+
+  const setSourceViewportFeaturesReady = useCallback(
+    (ready) => {
+      if (sourceId) {
+        dispatch(setViewportFeaturesReady({ sourceId, ready }));
+      }
+    },
+    [dispatch, sourceId]
+  );
+
+  const computeFeaturesGeoJson = useCallback(
+    ({ viewport, uniqueIdProperty }) => {
+      executeTask(sourceId, Methods.VIEWPORT_FEATURES_GEOJSON, {
+        viewport: Float32Array.of(...viewport),
+        uniqueIdProperty
+      })
+        .then(() => {
+          setSourceViewportFeaturesReady(true);
+        })
+        .catch(throwError);
```

Comment:
Previously we had a specific error case that was ignored: 
`if (error.name === 'AbortError') return;` and it has been removed, why?

Reply:
Also, why changing modern await & try catch with classic thenable?

Reply:
Not removed, just it was generalised:

```
export function throwError(error) {
  if (error.name === 'DataCloneError')
    throw new Error(
      `DataCloneError: Unable to retrieve GeoJSON features. Please check that your query is returning a column called "geom" or that you are using the geoColumn property to indicate the geometry column in the table.`
    );
  if (error.name === 'AbortError') return;
  throw error;
}
```

You can find it in [utils.js](https://github.com/CartoDB/carto-react/pull/238/files#diff-a6dd04cfab907cd4586bf8992ef06a0f6de2107df277078f0d2d0c2431ca7ace).

Reply:
At least for understanding, try catch option is only better when you have multiple await or many lines inside try. With just one promise and using just a few lines, I think try catch makes the code worse to follow and needs more lines than thenable option to make the same.

somethingAsync().then(...).catch(...)

It's better, at least for me, than:

try {
     const res = await somethingAsync()
     doSomething(res)
} catch (error) {
     ...
}

Btw, I don't think it's more modern, it's just another option that has its good/bad use cases like any other option.